dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -3808,6 +3847,8 @@ RtpsUdpDataLink::RtpsWriter::gather_heartbeats(OPENDDS_VECTOR(TransportQueueElem
     }
     readers_expecting_heartbeat_.clear();
   }
+
+  return !additional_guids.empty() || !preassociation_readers_.empty() || !lagging_readers_.empty() || !readers_expecting_heartbeat_.empty();
```

Comment:
Assuming these structures haven't changed since the top of the method, can this just return true? Not necessarily a suggestion, just a question.

Reply:
`readers_expecting_heartbeat_` may change so I think we need to check each condition.